### ISSUE TITLE
Cannot get current task information

### ISSUE THREAD

**Issue by DoctorDimon** [actionhero/actionhero#1175]
Hello!

I try to write a middleware that will monitor tasks, but I can get current task information. Here is my middleware initializer:

```
'use strict';
const {api, Initializer} = require('actionhero');

module.exports = class tasksFailedControl extends Initializer {
    constructor() {
        super();
        this.name = 'tasksFailedControl';
        this.loadPriority = 1100;
        this.startPriority = 1100;
        this.stopPriority = 1100;
    }

    async initialize() {

        const middleware = {
            name: 'tasksFailedControl',
            global: true,
            priority: 90,
            preProcessor: async () => {

                console.log(this);

            },
            postProcessor: async () => {
                let failedCount = await api.tasks.failedCount();
                let failedTasks = await api.tasks.failed(0, failedCount);
                for(let task of failedTasks) {

                    await api.tasks.removeFailed(task);
                    await api.tasks.enqueueAt((Date.now() + 10 * 1000), task.payload.class, {}, 'high');

                }
            }
        };
        api.tasks.addMiddleware(middleware)
    }
};
```

and here is the output of ` console.log(this);`:

`tasksFailedControl {
  name: 'tasksFailedControl',
  loadPriority: 1100,
  startPriority: 1100,
  stopPriority: 1100 }`

Please, help me find out, how to get current task information in middleware, thank you!

**Comment by evantahler** [comment]
It might be helpful to hear more about what you are trying to accomplish.   The code above will:
* before any task is run, log "this", which is just a description of the middleware (as you see).
* after any task is run, load all failed tasks in the system
  * If any failed task is found, remove it from the failed queue and re-enqueue it removing any params.

Is that what you intend?    If you, perhaps, want to retry all failed tasks, there are other ways to accomplish it.  You might be trying to read "this" in the task context, which you can't us anonymous functions for.  Try:

```js
const middleware = {
            name: 'tasksFailedControl',
            global: true,
            priority: 90,
            preProcessor: async function () { // note the "FUNCTION" 
                console.log(this);
            },
```

https://docs.actionherojs.com/tutorial-middleware.html

**Comment by DoctorDimon** [comment]
Thank you for your answer! Actually I want to retry all tasks witch was failed but not now, in 10 seconds from now. My tasks has no params, so i skip it.

What im trying to do - is to access task name in middleware and write stats from this task to DB.

Will try your example...

**Comment by DoctorDimon** [comment]
Yes it works!
Thank you! The problem was that i used `async () =>` insted of `async function()`. Actually I always think it is the same... Good experience!
Thanks again!
